test/unit/test_disco_deploy.py

```diff
@@ -1273,6 +1382,38 @@ def test_test_wo_amis(self):
         self._ci_deploy.test()
         self.assertEqual(self._ci_deploy.test_ami.call_count, 0)
 
+    def test_test_with_restrict_ami(self):
+        '''Test test with spcified restrict amis'''
+        self._ci_deploy._restrict_amis = [self._amis_by_name['mhcbar 2'].id]
+        amis = [self._amis_by_name['mhcbar 2']]
+        self._ci_deploy._disco_bake.list_amis = MagicMock(return_value=amis)
+        self._ci_deploy.get_test_amis = MagicMock(return_value=[])
+        self._ci_deploy.test_ami = MagicMock()
+        self._ci_deploy.test()
+        self.assertEqual(self._ci_deploy.get_test_amis.call_count, 0)
+        self._ci_deploy._disco_bake.list_amis.assert_called_with(ami_ids=[self._amis_by_name['mhcbar 2'].id])
+        self.assertEqual(self._ci_deploy.test_ami.call_count, 1)
+
+    def test_test_with_invalid_restrict_ami(self):
+        '''Test test with spcified restrict amis'''
+        self._ci_deploy._restrict_amis = [self._amis_by_name['mhcbar 2'].id]
+        self._ci_deploy._disco_bake.list_amis = MagicMock(return_value=[])
+        self._ci_deploy.get_test_amis = MagicMock(return_value=[])
+        self._ci_deploy.test_ami = MagicMock()
+        self._ci_deploy.test()
+        self.assertEqual(self._ci_deploy.get_test_amis.call_count, 0)
+        self._ci_deploy._disco_bake.list_amis.assert_called_with(ami_ids=[self._amis_by_name['mhcbar 2'].id])
+        self.assertEqual(self._ci_deploy.test_ami.call_count, 0)
+
+    def test_test_wo_restrict_ami(self):
+        '''Test test with spcified restrict amis'''
```

Comment:
This docstring doesn't seem to agree with the name of the method. Also, wouldn't we want to verify that it uses the latest ami?